### PR TITLE
feat: Extend logic to detect charms from charmcraft.yaml

### PR DESCRIPTION
**Pull request by kimwnasptd**
This came up while working on https://github.com/canonical/github-profiles-automator/issues/7

The newer version of `charmcraft init` now only creates a `charmcraft.yaml` file and not a `metadata.yaml` one
https://juju.is/docs/sdk/metadata-yaml

## Changes Made
1. Update the format of the code
    1. Run `black` on the touched python script
    2. Resolve typing errors, since `base_dir` was used as a `str` and a `Path()` value
2. Extend the code and tests to detect charm dirs if a `charmcraft.yaml` file is present
## Review Notes
I didn't take into account a potential `charmcraft.yml` file since `charmcraft init` only initialises files with `.yaml` extension.

### CODE REVIEW

File: actions/get-charm-paths/src/get_charm_paths.py
Comment (orfeas-k) [inline actions/get-charm-paths/src/get_charm_paths.py:1]: Should the commited file have the proper permissions in order to be used by the workflow? Maybe not, but I had to `chmod +x` in order to test it.
```
~/github/kubeflow-ci ‹ac10794› [uk8s:kubeflow]
$ l actions/get-charm-paths/src 
total 12K
drwxrwxr-x 2 orfeas orfeas 4.0K Nov 28 14:25 .
drwxrwxr-x 4 orfeas orfeas 4.0K Nov 17  2023 ..
-rw-rw-r-- 1 orfeas orfeas 2.8K Nov 28 14:25 get_charm_paths.py
-rw-rw-r-- 1 orfeas orfeas    0 Nov 17  2023 __init__.py
```

File: actions/get-charm-paths/src/get_charm_paths.py [hunk 1]
```diff
@@ -3,59 +3,86 @@
...
     """
-    base_dir = Path(base_dir)
-    charms_dir = base_dir / charms_subdir
-    
-    metadata_files = list(charms_dir.glob("*/metadata.yaml")) + list(charms_dir.glob("*/metadata.yml"))
-    
+    root_dir = Path(base_dir)
+    charms_dir = root_dir / charms_subdir
+
+    metadata_files = (
```
Comment (orfeas-k) [inline actions/get-charm-paths/src/get_charm_paths.py:28]:
```suggestion
    tracked_files = (
```
? 
or sth like that

File: actions/get-charm-paths/src/get_charm_paths.py [hunk 2]
```diff
@@ -3,59 +3,86 @@
...
+        + list(charms_dir.glob("*/metadata.yml"))
+        + list(charms_dir.glob("*/charmcraft.yaml"))
+    )
+
     if not metadata_files:
-        # No nested charm directories found, check if the top level dir is a charm
-        metadata_files = list(base_dir.glob("metadata.yaml")) + list(base_dir.glob("metadata.yml"))
+        # No nested charm directories found, check if the top level dir is
+        # a charm
+        metadata_files = (
```
Comment (orfeas-k) [inline actions/get-charm-paths/src/get_charm_paths.py:37]: same

File: actions/get-charm-paths/src/get_charm_paths.py
Comment (kimwnasptd) [reply actions/get-charm-paths/src/get_charm_paths.py:1]: Hmm interesting, for sure let's add it.

Was using `python3 ...` to trigger the script and didn't catch it de10c11

File: actions/get-charm-paths/src/get_charm_paths.py (hunk 1 above)
Comment (kimwnasptd) [reply actions/get-charm-paths/src/get_charm_paths.py]: Sent a commit 322e2be

File: actions/get-charm-paths/src/get_charm_paths.py (hunk 2 above)
Comment (kimwnasptd) [reply actions/get-charm-paths/src/get_charm_paths.py]: Sent a commit 322e2be

File: actions/get-charm-paths/src/get_charm_paths.py
Comment (orfeas-k) [reply actions/get-charm-paths/src/get_charm_paths.py:1]: you 're right actually, my bad https://github.com/canonical/kubeflow-ci/blob/main/actions/get-charm-paths/action.yml#L26

**Pull request merged**